utils/fixed_decimal/src/decimal.rs

```diff
@@ -2078,6 +2078,185 @@ impl FixedDecimal {
+            if significand_str.contains(&b'E')
+                || exponent_str.iter().any(|&c| c == b'e' || c == b'E')
+            {
+                return Err(Error::Syntax);
+            }
+            let exponent = FixedDecimal::try_from(exponent_str)?;
+            ScientificDecimal::from_parts(significand, exponent)
+        } else {
+            Err(Error::Syntax)
+        }
+    }
+}
+
+/// A struct containing a [`FixedDecimal`] significand together with an exponent, representing a
+/// number written in compact notation (such as 1.2M).
+/// This represents a _source number_ that uses compact decimal notation, as defined
+/// [in UTS #35](https://www.unicode.org/reports/tr35/tr35-numbers.html#Plural_rules_syntax).
+///
+/// This is distinct from [`ScientificDecimal`] because it does not represent leading 0s
+/// nor a sign in the exponent, and behaves differently in pluralization.
+#[derive(Debug, Clone, PartialEq)]
+pub struct CompactDecimal {
+    significand: FixedDecimal,
+    exponent: i16,
+}
+
+/// Render the [`CompactDecimal`] in sampleValue syntax.
+/// The letter c is used, rather than the deprecated e.
+///
+/// # Examples
+///
+/// ```
+/// # use fixed_decimal::CompactDecimal;
+/// # use std::str::FromStr;
+/// # use writeable::assert_writeable_eq;
+/// #
+/// assert_writeable_eq!(CompactDecimal::from_str("+1.20c6").unwrap(), "+1.20c3");
+/// ```
+impl writeable::Writeable for CompactDecimal {
+    fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
+        self.significand.write_to(sink)?;
+        sink.write_char('c')?;
+        self.exponent.write_to(sink)
+    }
+
+    fn writeable_length_hint(&self) -> writeable::LengthHint {
+        self.significand.writeable_length_hint() + 1 + self.exponent.writeable_length_hint()
+    }
+}
+
+writeable::impl_display_with_writeable!(CompactDecimal);
+
+impl FromStr for CompactDecimal {
+    type Err = Error;
+    fn from_str(input_str: &str) -> Result<Self, Self::Err> {
+        Self::try_from(input_str.as_bytes())
+    }
+}
+
+/// The deprecated letter e is not accepted as a synonym for c.
+impl TryFrom<&[u8]> for CompactDecimal {
+    type Error = Error;
+    fn try_from(input_str: &[u8]) -> Result<Self, Self::Error> {
+        if input_str.iter().any(|&c| c == b'e' || c == b'E') {
+            return Err(Error::Syntax);
+        }
+        if let Some((significand_str, mut exponent_str)) = input_str
+            .iter()
+            .position(|&c| c == b'c')
+            .map(|i| input_str.split_at(i))
+        {
+            let significand = FixedDecimal::try_from(significand_str)?;
+            #[allow(clippy::indexing_slicing)] // Nonempty, contains at least 'c'.
+            {
+                exponent_str = &exponent_str[1..]; // Skip the 'c'.
+            }
+            if exponent_str.is_empty() {
+                return Err(Error::Syntax);
+            }
+            #[allow(clippy::indexing_slicing)] // Nonempty.
```

Comment:
Suggestion: It's best if you can write your code in a way such that you don't need these suppressions. In addition to being cleaner, doing so can help prove to the compiler that it doesn't need to branch for the panic.